quinn/src/platform/unix.rs

```diff
@@ -146,7 +146,7 @@ impl super::UdpExt for UdpSocket {
     fn send_ext(&self, transmits: &[Transmit]) -> io::Result<usize> {
         let mut msgs: [libc::mmsghdr; BATCH_SIZE] = unsafe { mem::zeroed() };
         let mut iovecs: [libc::iovec; BATCH_SIZE] = unsafe { mem::zeroed() };
-        let mut cmsgs = [cmsg::Aligned(MaybeUninit::uninit()); BATCH_SIZE];
+        let mut cmsgs = [cmsg::Aligned([0u8; CMSG_LEN]); BATCH_SIZE];
```

Comment:
FYI: This can't be uninit because it breaks `CMSG_NXTHDR`. The function will try to access the next cmsg in the buffer at https://github.com/rust-lang/libc/blob/ae65df55fe2d0e1348fed575bfc8c0c3e4744c65/src/unix/linux_like/linux/mod.rs#L2585-L2589

This will yield an undefined result, and can let `CMSG_NEXT` return `None` - which allows for no message except the first one to be written. A different approach is to  manually adjust the cmsg offset during encoding - but I think sticking to the official macros seems safer.

Reply:
This was btw really painful to debug

Reply:
Oh geez, nice catch!